rest/tests/test_rest.py

```diff
@@ -524,9 +532,34 @@ def test_immutable_fields(self):
         self.assertNotEqual(vocab_dict['slug'], result['slug'])
         self.assertNotIn('repository', result)
         vocabulary = Vocabulary.objects.get(slug=result['slug'])
-        self.assertEqual(repository.id, vocabulary.repository.id)
-        vocab_dict = result
+        self.assertEqual(self.repo.id, vocabulary.repository.id)
+        vocab_slug = result['slug']
+
+        result = self.patch_vocabulary(
+            self.repo.slug, vocab_slug, vocab_dict, skip_assert=True)
+        # these keys should be different since they are immutable or set by
+        # the serializer
+        self.assertNotEqual(vocab_dict['id'], result['id'])
+        self.assertNotEqual(vocab_dict['slug'], result['slug'])
+        self.assertNotIn('repository', result)
+        vocabulary = Vocabulary.objects.get(slug=result['slug'])
+        self.assertEqual(self.repo.id, vocabulary.repository.id)
+        vocab_slug = result['slug']
 
+        result = self.put_vocabulary(
```

Comment:
This set of asserts is in this test three times. I could be a good use for a nested function.